src/model/relationship/RelationshipImpl.java

```diff
@@ -0,0 +1,190 @@
+      return RelationshipType.platonicType(regard);
+    } else {
+      return this.romanticType;
+    }
+  }
+
+  @Override
+  public boolean isRomantic() {
+    return romanticType != null;
+  }
+
+  private boolean rollFor(Random r, double chance) {
+    return r.nextDouble() < chance;
+  }
+
+  @Override
+  public void progressRelationship(Random r) {
+    progressRegard(r);
+    if (rollFor(r, progressDesireChance(r))) {
+      growDesire();
+      if (this.isRomantic()) {
+        if (rollFor(r, progressRomanticTypeChance())) {
+          progressRomanticType();
+        }
+      } else {
+        if (rollFor(r, startDatingChance())) {
+          progressRomanticType();
+        }
+      }
+    } else {
+      lessenDesire();
+      if (this.isRomantic()) {
+        if (rollFor(r, breakUpChance())) {
+          breakUp();
+        }
+      }
+    }
+  }
+
+  private void progressRegard(Random r) {
+    if (rollFor(r, progressRegardChance(r))) {
+      growRegard();
+    } else {
+      lessenRegard();
+    }
+  }
+
+  private double progressRegardChance(Random r) {
+    return r.nextDouble(); // TODO: add some actual logic
+  }
+
+  private double startDatingChance() {
+    if (this.p1.hasSignificantOther() || this.p2.hasSignificantOther() ||
+        this.p1.isMourningSpouse() || this.p2.isMourningSpouse()) {
+      return 0;
+    } else {
+      return progressRomanticTypeChance();
+    }
+  }
+
+  private double progressDesireChance(Random r) {
+    return r.nextDouble(); // TODO: add some actual logic
+  }
+
+  private void progressRomanticType() {
+    switch (this.getType()) {
+      case FRIEND:
+        this.romanticType = RelationshipType.PARTNER;
+        this.p1.setSignificantOther(p2);
+        p2.setSignificantOther(p1);
+        break;
+      case PARTNER:
+        this.romanticType = RelationshipType.FIANCE;
+        break;
+      case FIANCE:
+        this.romanticType = RelationshipType.SPOUSE;
+        break;
+      default:
+        break;
+    }
+  }
+
+  private double progressRomanticTypeChance() {
+    double upperThreshold = (this.isRomantic()) ? this.getType().getUpperThreshold() : RelationshipType.PARTNER.getLowerThreshold();
+    if (this.desire <= upperThreshold) {
+      return 0;
+    } else {
+      return this.desire; //TODO: add some actual logic
+    }
+  }
+
+  private double breakUpChance() { // should this also include a random factor?
+    if (this.isRomantic()) {
+      double reasonsToStay  = this.desire + this.regard /2; // TODO: add how long they've been together
+      double total = 1.5;
+      return total - reasonsToStay;
+    } else {
+      return  0;
+    }
+
+  }
+
+  private void romanticToPlatonic() {
+    this.romanticType = null;
+  }
+
+  private void breakUp() {
+    this.p1.setSignificantOther(null);
+    this.p2.setSignificantOther(null);
+    this.romanticToPlatonic();
+    // TODO: isn't there a list of past spouses/significant others? if so, this add to that
```

Comment:
When will this TODO be handled?

Reply:
now

Reply:
Ah okay. If it's already been handled let's remove this TODO.